charts/connect/templates/operator-pdb.yaml

```diff
@@ -9,13 +9,14 @@ metadata:
     {{- include "onepassword-connect.labels" . | nindent 4 }}
 {{- include "onepassword-connect.extraAnnotations" .Values.operator.pdb }}
 spec:
-  {{- with .Values.operator.pdb.maxUnavailable }}
-  maxUnavailable: {{ . }}
-  {{- else }}
+  {{- if .Values.operator.pdb.minAvailable }}
   minAvailable: {{ .Values.operator.pdb.minAvailable }}
   {{- end }}
+  {{- if and .Values.operator.pdb.maxUnavailable (not .Values.operator.pdb.minAvailable)}}
+  maxUnavailable: {{ .Values.operator.pdb.maxUnavailable }}
+  {{- end }}
```

Comment:
`[question]`

Looking at this it seems that we will always prioritise setting `minAvailable` over `maxUnavailable` if both are accidentally provided (which ideally shouldn't happen). Is there any case in which this assumption will become problematic? 🤔 

By looking at [the documentation](https://kubernetes.io/docs/tasks/run-application/configure-pdb/), it seems that `maxUnavailable` might be the recommended one to use:
> The use of maxUnavailable is recommended as it automatically responds to changes in the number of replicas of the corresponding controller.

And by looking at the way [this configuration is implemented](https://github.com/kubernetes/kubernetes/blob/23be9587a0f8677eb8091464098881df939c44a9/pkg/controller/disruption/disruption.go#L539), it seems that `maxUnavailable` take priority over `minAvailable`.

---

Another question: How is this different from previous implementation?

From my understanding the differences are the following:
- Previous implementation set `maxUnavailable` for each Helm deployment. This new version only sets it if it's specifically provided.
- Previous implementation prioritized `maxUnavailable` over `minAvailable`. This new version does the opposite.
- Both configurations only set one of the two values. The isn't a case in which both are set.

Reply:
>Is there any case in which this assumption will become problematic?

I don't think so, but as an alternative, we can remove the if conditions and delegate the responsibility to the final user receiving an error from the API like the following:
```
Invalid value: policy.PodDisruptionBudgetSpec{MinAvailable:(*intstr.IntOrString)(0xc01c719c00), Selector:(*v1.LabelSelector)(0xc01c719c20), MaxUnavailable:(*intstr.IntOrString)(0xc01c719cc0), UnhealthyPodEvictionPolicy:(*policy.UnhealthyPodEvictionPolicyType)(nil)}: minAvailable and maxUnavailable cannot be both set
```

> It seems that maxUnavailable might be the recommended

The example uses a StatefulSet, for which I also recommend using the `maxUnavailable` because we want to move one pod at a time. However, this helm chart uses Deployments for both `connect` and `operator`. In the case of Deployments, I prefer the use of the `minAvailable ` because we need only one running at a time to keep the service running.


> It seems that maxUnavailable takes priority over `minAvailable`.

It depends on your point of view. It is an if/else condition. In my opinion, both have the same priority.

>About differences

I agree with all the sentences, but the main difference, IMO, is that now we can set `minAvailable` or `maxUnavailable`.

Reply:
> It depends on your point of view. It is an if/else condition. In my opinion, both have the same priority.

Indeed, based on the implementation of the PR, both have the same priority. I was looking at how these configuration values are [parsed by Kubernetes in code](https://github.com/kubernetes/kubernetes/blob/23be9587a0f8677eb8091464098881df939c44a9/pkg/controller/disruption/disruption.go#L526-L566) to understand the priority. And it looks like if _by accident_ you set both, `maxUnavailable` takes priority, while `minAvailable` will be ignored.

With that in mind and attempting to make this change change as little behavior as possible, I'm leaning towards us doing the same in the Helm chart. Therefore, I might recommend here swapping the conditionals: check if `maxUnavailable` is set with simple if, and for `minAvailable` if it's set _and_ `maxUnavailable` is _not_ set.

Let me know how you feel about this.

Reply:
> > Is there any case in which this assumption will become problematic?
> 
> I don't think so, but as an alternative, we can remove the if conditions and delegate the responsibility to the final user receiving an error from the API.

This sounds like a good alternative. Another option is to document that one takes priority over the other and give a warning when both are set. I'm personally ok with either what's now in the PR, your alternative solution or just give a warning when both are set.

Reply:
Reverted